tex/latex/pgf-pie/tikzlibrarypie.code.tex

```diff
@@ -205,6 +205,10 @@
 }
 
 \def\pgfpie@@pie[#1]#2{%
+  \begingroup
+  % to be compatible with "babel" lib
+  \tikzset{handle active characters in nodes=false}
+
   % load default parameters
   \pgfqkeys{/pgfpie}{
```

Comment:
I think we could do a scope here and fuse it with the subsequent `\pgfqkeys`, somewhat like this
```suggestion
  \scope[
    handle active characters in nodes=false,
    /pgfpie/.cd,
```

Reply:
What do you think of this?
```patch
diff --git a/tex/latex/pgf-pie/tikzlibrarypie.code.tex b/tex/latex/pgf-pie/tikzlibrarypie.code.tex
index 3d62a01..6caf677 100644
--- a/tex/latex/pgf-pie/tikzlibrarypie.code.tex
+++ b/tex/latex/pgf-pie/tikzlibrarypie.code.tex
@@ -154,6 +154,7 @@
 \newcount\pgfpie@sliceLength
 
 \pgfqkeys{/pgfpie}{%
+  .search also={/tikz,/pgf},
   explode/.store in=\pgfpie@explode,
   color/.store in=\pgfpie@color,
   radius/.store in=\pgfpie@radius,
@@ -162,7 +163,12 @@
   before number/.store in=\pgfpie@beforenumber,
   after number/.store in=\pgfpie@afternumber,
   text/.store in=\pgfpie@text,
-  sum/.store in=\pgfpie@sum,
+  sum/.code={
+    \def\pgfpie@sum{#1}%
+    \pgfpie@ifx\pgfpie@sum\pgfpie@sum@c{%
+      \def\pgfpie@afternumber{\%}%
+    }{}%
+  },
   rotate/.store in=\pgfpie@rotate,
 }
 
@@ -205,12 +211,9 @@
 }
 
 \def\pgfpie@@pie[#1]#2{%
-  \begingroup
-  % to be compatible with "babel" lib
-  \tikzset{handle active characters in nodes=false}
-
-  % load default parameters
-  \pgfqkeys{/pgfpie}{
+  \scope[%
+    handle active characters in nodes=false,
+    /pgfpie/.cd,
     explode=0,
     color={blue!60, cyan!60, yellow!60, orange!60, red!60,
       blue!60!cyan!60, cyan!60!yellow!60, red!60!cyan!60,
@@ -230,14 +233,7 @@
     hide number=false,
     hide label=false,
     every pie/.try,
-  }%
-  % load user's parameters
-  \pgfqkeys{/pgfpie}{#1}%
-  % add percentage automatically
-  \pgfpie@ifx\pgfpie@sum\pgfpie@sum@c{%
-    \pgfqkeys{/pgfpie}{after number=\%}%
-    \pgfqkeys{/pgfpie}{#1}%
-  }{}%
+    #1]
 
   % legend or not
   \pgfpie@ifx\pgfpie@text\pgfpie@text@legend{%
@@ -284,7 +280,8 @@
     \pgfpie@iflegend{%
       \pgfpie@legend{#2}%
   }{}}%
-  \endgroup
+
+  \endscope
 }
 
 \def\pgfpie@pie@square#1{%
```

Reply:
```diff
-  sum/.store in=\pgfpie@sum,
+  sum/.code={
+    \def\pgfpie@sum{#1}%
+    \pgfpie@ifx\pgfpie@sum\pgfpie@sum@c{%
+      \def\pgfpie@afternumber{\%}%
+    }{}%
+  },
```
This will change the current behavior. For example, with `\pie[sum=auto]` (noted `sum=100` is always called before `\pie`'s opt-arg), currently `\pgfpie@afternumber` is empty, with the suggested change it will be set to `\%`.

Executing `\pie`'s opt-arg twice is most likely wrong, and how `sum=100` should interact with the following user provided `sum=<val not 100>` or `after number=<val>` is another problem. How about
```tex
\pgfqkeys{/pgfpie}{
  after number/.code={%
    \def\pgfpie@afternumber{#1}%
    \def\pgfpie@late@afternumber{}%
  },
  sum/.code={%
    \def\pgfpie@sum{#1}%
    \pgfpie@ifx\pgfpie@sum\pgfpie@sum@c{%
      \def\pgfpie@late@afternumber{\def\pgfpie@afternumber{\%}}%
    }{%
      \def\pgfpie@late@afternumber{}%
    }%
  },
}

\pgfpie@@pie[#1]#2{%
  \pgfkeysq{/pgfpie}{..., after command=, sum=100, ..., #1}%
  \pgfpie@late@afternumber
  % remaining
}
```

The `\scope` part is OK.

Reply:
pin

Reply:
LGTM